test/evp_test.c

```diff
@@ -1258,6 +1259,13 @@ static int pkey_test_ctrl(EVP_PKEY_CTX *pctx, const char *value)
     if (p != NULL)
         *p++ = 0;
     rv = EVP_PKEY_CTX_ctrl_str(pctx, tmpval, p);
+    if (p != NULL && rv <= 0 && rv != -2) {
```

Comment:
If there's an actual typo, such as `Ctrl = digest:MCD2`, this check should still fail, since MCD2 doesn't have an OID...